src/main/java/fm/last/moji/local/LocalMojiFileAttributes.java

```diff
@@ -48,4 +48,9 @@ public String getKey() {
     return mojiFile.getKey();
   }
 
+  @Override
+  public String getChecksum() {
+    return "";
```

Comment:
Aah yes, sorry, I misunderstood this (it's been a while since I looked at this code and I had to drag @teabot over to help remove the rust from my brain ;) ). The "local" classes as you've probably seen are mainly just for unit testing on a local file system. We could have this method use the getInputStream() from mojiFile and then run this through a checksum algorithm and return a real checksum? The downside is that this will take time to perform the calculation but it's probably fine if we limit how many times we call this in tests (and the size of the underlying files). So, it's up to you, either implement a checksum or leave this returning null with a comment suggesting this as a future enhancement. The rest of your changes look good.